src/olympia/devhub/views.py

```diff
@@ -406,6 +406,10 @@ def disable(request, addon_id, addon):
 def unlist(request, addon_id, addon):
     addon.update(is_listed=False, disabled_by_user=False)
     amo.log(amo.LOG.ADDON_UNLISTED, addon)
+
+    if addon.latest_version.is_unreviewed:
+        auto_sign_version(addon.latest_version)
```

Comment:
The developer won't get any notification (e.g. email) using this method btw - it's used currently only for API uploads.